### ISSUE TITLE
compile issue on macOS for version v0.10.0

### ISSUE THREAD

**Issue by slimdevl** [enriquebris/goworkerpool#39]
First let me say...awesome library, extremely useful!

On Mac I am getting strange compile error for v0.10.0 which I didn't see on Linux:

```bash
$ go get github.com/enriquebris/goworkerpool@v0.10.0
# github.com/enriquebris/goworkerpool
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:166:102: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.totalWorkers.SetTriggerOnValue
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:168:60: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.totalWorkers.EnqueueToRunAfterCurrentTriggerFunctions
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:201:64: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.totalWorkers.SetTriggerOnValue
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:903:64: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.totalWorkers.SetTriggerOnValue
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:905:60: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.totalWorkers.EnqueueToRunAfterCurrentTriggerFunctions
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:1022:68: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.taskSuccesses.SetTriggerOnValue
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:1027:61: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.taskSuccesses.EnqueueToRunAfterCurrentTriggerFunctions
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:1053:80: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.totalWorkersInProgress.SetTriggerOnValue
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:1055:70: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.totalWorkersInProgress.EnqueueToRunAfterCurrentTriggerFunctions
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:1070:61: cannot use func literal (type func()) as type goconcurrentcounter.concurrentIntFunc in argument to st.taskSuccesses.EnqueueToRunAfterCurrentTriggerFunctions
../../go/pkg/mod/github.com/enriquebris/goworkerpool@v0.10.0/pool.go:1070:61: too many errors
```

Looking at the `Int` interface, indeed the signature is wrong.
https://github.com/enriquebris/goworkerpool/blob/master/pool.go#L166
vs 
https://github.com/enriquebris/goconcurrentcounter/blob/master/concurrentCounter.go#L4

**Comment by enriquebris** [comment]
@slimdevl Good catch, I need to fix it ASAP. I'll let you know as soon as I deliver the new version (having this fixed). Already working on it ...

**Comment by enriquebris** [comment]
@slimdevl The compile issue is fixed in [v0.10.1](https://github.com/enriquebris/goworkerpool/releases/tag/v0.10.1) and in [latest version: v0.10.2](https://github.com/enriquebris/goworkerpool/releases/tag/v0.10.2).

```bash
$ go get github.com/enriquebris/goworkerpool
go get: added github.com/enriquebris/goconcurrentcounter v0.0.0-20220112230718-926268af96e3
go get: added github.com/enriquebris/goconcurrentqueue v0.6.0
go get: added github.com/enriquebris/goworkerpool v0.10.2
```

Thanks for your interest in this package!

**Comment by ghost** [comment]
Works great! Thanks for the fast turn around 💯
